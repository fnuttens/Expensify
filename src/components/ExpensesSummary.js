import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

const formatSummary = (count, total) => {
  const pluralExpense = count > 1 ? 'expenses' : 'expense';
  const totalAmount = numeral(total / 100).format('$0,0.00');
  return `Viewing ${count} ${pluralExpense} totalling ${totalAmount}`;
};

export const ExpensesSummary = (props) => (
  <div>
    <p>{formatSummary(props.expenseCount, props.expensesTotal)}</p>
  </div>
);

const mapStateToProps = (state) => ({
  expenseCount: state.expenses.length,
  expensesTotal: selectExpensesTotal(selectVisibleExpenses(state.expenses, state.filters)),
});

export default connect(mapStateToProps)(ExpensesSummary);
