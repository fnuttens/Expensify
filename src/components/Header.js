import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink exact to="/" activeClassName="is-active">Dashboard</NavLink><br />
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br />
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
);

export default Header;
