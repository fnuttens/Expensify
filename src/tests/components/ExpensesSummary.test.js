import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render 1 expense correctly', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={1}
      expensesTotal={expenses[1].amount}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render several expenses correctly', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={2}
      expensesTotal={114000}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
