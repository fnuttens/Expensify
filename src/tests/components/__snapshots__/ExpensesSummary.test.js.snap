// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense correctly 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $1,095.00
  </p>
</div>
`;

exports[`should render several expenses correctly 1`] = `
<div>
  <p>
    Viewing 2 expenses totalling $1,140.00
  </p>
</div>
`;
